server/jetstream_api.go

```diff
@@ -1824,18 +1828,32 @@ func (s *Server) jsStreamInfoRequest(sub *subscription, c *client, a *Account, s
 
 	// Check if they have asked for subject details.
 	if subjects != _EMPTY_ {
-		if mss := mset.store.SubjectsState(subjects); len(mss) > 0 {
-			if len(mss) > JSMaxSubjectDetails {
-				resp.StreamInfo = nil
-				resp.Error = NewJSStreamInfoMaxSubjectsError()
-				s.sendAPIErrResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(&resp))
-				return
+		if mss := mset.store.SubjectsState(subjects); len(mss) > 0 && offset < len(mss) {
+			// As go iterates over map in a non-consistent order, no choice but to buffer it a slice
+
+			buffer := make([]string, 0, len(mss))
+			for subj := range mss {
+				buffer = append(buffer, subj)
+			}
+
+			// Sort it
+			sort.Strings(buffer)
+
+			limitSize := JSMaxSubjectDetails
```

Comment:
Move these 2 lines below computation of offset and end, and you could have:
```
var sd map[string]uint64
limitSize := end - offset
if limitSize > 0 {
   sd = make(map[string]uint64, limitSize)
   for _, ss := range buffer[offset:end] {
   ...
}
resp.StreamInfo.State.Subjects = sd
...
```

Reply:
I see what you mean, implemented the optimization thanks for the suggestion